vcardtools.py

```diff
@@ -111,6 +116,25 @@ def init_parser():
     )
     return parser
 
+def sanitise_name(a_name: str) -> str:
+    """ Sanitise the name, basically a filename,
+        by removing characters which would cause a problem when creating a file in the OS
+        and replacing them with something safe (in this case, an underscore)
+    """
+    NEW_REPLACEMENT_CHAR = '_'
+    FROM_CHARACTERS = ' .\\/"\'!@#?$%^&*|()[]{};:<>'
+    for old_char in FROM_CHARACTERS:
+        a_name = a_name.replace(old_char, NEW_REPLACEMENT_CHAR)
+
+    # An optional extra would be to remove all duplicates of the underscore
+    return a_name
+
+def mk_vcard_fnm(a_name: str = '', ext: str = '') -> str:
```

Comment:
Hey @tonysutherland,
Thank you for your feedback, and sorry it took that long for me to reply ... very busy !
Ahah semantics is always tricky :wink:
For me, when I read `mk_x` I think about actually impacting the filesystem, cause I am so used to `mkfs` and `mktemp` and so on :sweat_smile:
At the same time I intuitively agree with you about `get` == `getter`, but in most source code I have read it is also used to *build/generate/retrieve* something. So I am having mixed feelings here.
One thing I am sure of is that I would rather have written `x_filename` than `x_fnm` because it is more explicit and hence more maintainable.
What about `generate_vcard_filename` ? It is even more explicit.
And same for the group: `generate_group_dirname`...
What do you say ?
Thanks in advance

Reply:
I completely understand about being rather busy.  Sort of same here. Can be difficult to find time to deal with many of these "side" projects.
I will try to make everything as time efficient as possible.
Before I do the final code, I think it is best to resolve some of these issues. I am not too fussed about the timeframe, since I have already done what I needed to do with the vcardtools code anyway.

With respect to your suggestion of using the verb "generate", it is interesting that I use that everywhere in my code.  And I tend to use the full "filename" in many places, but occasionally "fnm" (only because it is relatively common).  I have also taken to using "folder" and "foldername" rather than "dir" or "directory", but I am sanguine either way. "Dirname" is fine by me.

I think "generate" is much more explicit and semantically correct.
It then makes sense that "mk" is used for persistent storage type calls (as you said, mkfs, mktemp, mkdirs, etc); "get" and "set" are used for object property getting and setting and "generate" is used for generating/creating a new thing.

I will use generate_vcard_filename and generate_group_dirname.

Reply:
I have done the renaming in a9be502
And added the group dirname function in 50583d2